### PR TITLE
Add ritual initialization to client

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
- Feature

**Required reviews:** 
- 2

**What this does:**
- Replaces hardcoded ritual with a proper ritual initialization in `DkgClient.initializeRitual`
- Implements accessory methods, such as waiting for ritual initialization to finish

**Issues fixed/closed:**
> - Fixes #...

**Why it's needed:**
- Currently, users may only use premade rituals and only under certain constraints. See https://github.com/nucypher/nucypher-ts/pull/232
- With this PR, users should be able to initialize rituals on their own using `Cohort` and `Strategy` APIs 

**Notes for reviewers:**
- Based on #232
- Tested on `lynx` with hand-picked `Cohort`. Documented a case where ritual initialization fails when `Cohort` doesn't meet staking requirements here: #235

### CODE REVIEW

File: src/sdk/strategy/cbd-strategy.ts [hunk 1]
```diff
@@ -30,15 +30,13 @@ export class CbdStrategy {
   }
 
   public async deploy(
-    web3Provider: ethers.providers.Web3Provider
+    web3Provider: ethers.providers.Web3Provider,
+    blockUntilDeployed = true
```
Comment (piotr-roslaniec) [inline src/sdk/strategy/cbd-strategy.ts:34]: Is this a sensible default?

Comment (manumonti) [reply src/sdk/strategy/cbd-strategy.ts:34]: It doesn't sound bad to me, but maybe I'm not the best person to say it.

**Review by manumonti — APPROVED** [review]
LGTM! 👏

**Review by manumonti — APPROVED** [review]

File: src/sdk/strategy/cbd-strategy.ts (hunk 1 above)
Comment (theref) [reply src/sdk/strategy/cbd-strategy.ts:34]: i like it, you don't want to get in the situation of attempting to use a ritual that never successfully completed

**Review by theref — APPROVED** [review]
Looks good!

**Pull request merged**